src/sentry/event_manager.py

```diff
@@ -377,7 +377,13 @@ def _normalize_impl(self, project_id: Optional[int] = None) -> None:
             **DEFAULT_STORE_NORMALIZER_ARGS,
         )
 
+        pre_normalize_type = self._data.get("type")
         self._data = CanonicalKeyDict(rust_normalizer.normalize_event(dict(self._data)))
+        # XXX: This is a hack to make generic events work (for now?). I'm not sure whether we should
+        # include this in the rust normalizer, since we don't want people sending us these via the
+        # sdk.
```

Comment:
Should we consult w/ the SDK team on this? I agree it seems like we wouldn't want this coming from the SDK though.

Reply:
Yeah, I'll bring it up with them, but I don't think it's a blocker for the moment since we're just reverting the event type here.